### PR TITLE
Add slack verification according to slack docs

### LINKED ISSUE #87: Verify slack token
We have a function `verify_token`, but it's never used.
So we don't actually verify that the request is from slack it seems. We should do that 😄

### PR DESCRIPTION
**Pull request by kamigerami**
Fixes slack verification according to instructions following the slack docs:

https://api.slack.com/docs/verifying-requests-from-slack

fixes #87

### CODE REVIEW

File: chalicelib/lib/slack.py [hunk 1]
```diff
@@ -144,13 +147,24 @@ def slack_payload_extractor(req):
         return "failed extracting payload", 200
 
 
-def verify_token(token):
-    if token == os.getenv('slack_token'):
+def verify_token(request, token):
+    #https://api.slack.com/docs/verifying-requests-from-slack
```
Comment (parberge) [inline chalicelib/lib/slack.py]: Remove this or add a real docstring with this information

File: chalicelib/lib/slack.py [hunk 2]
```diff
@@ -144,13 +147,24 @@ def slack_payload_extractor(req):
...
+
+    request_timestamp = request.headers['X-Slack-Request-Timestamp']
+
+    # create basestring out of v0 + timestamp + req_body
+    request_basestring = bytes('v0:' + request_timestamp + ':' + request, 'utf-8')
```
Comment (parberge) [inline chalicelib/lib/slack.py]: Can't this be with nice f-strings instead? (i.e `f"v0:{request_timestamp}:{request}"`)

File: chalicelib/lib/slack.py [hunk 3]
```diff
@@ -144,13 +147,24 @@ def slack_payload_extractor(req):
...
+    slack_signature = bytes(request.headers['X-Slack-Signature'], 'utf-8')
+    slack_token = token
```
Comment (parberge) [inline chalicelib/lib/slack.py]: Any point of this variable instead of just using `token` below?

File: chalicelib/lib/slack.py (hunk 1 above)
Comment (kamigerami) [reply chalicelib/lib/slack.py]: ok

File: chalicelib/lib/slack.py (hunk 2 above)
Comment (kamigerami) [reply chalicelib/lib/slack.py]: good point!

File: chalicelib/lib/slack.py (hunk 3 above)
Comment (kamigerami) [reply chalicelib/lib/slack.py]: no - we can use token directly! good point

File: tests/test_slack.py [hunk 4]
```diff
@@ -22,10 +22,13 @@ def test_slack_payload_extractor_payload():
 
 
 def test_slack_token():
-    assert verify_token('faulty fake token') is not True
-    fake_test_token = 'my_fake_token'
-    os.environ['slack_token'] = fake_test_token
-    assert verify_token(fake_test_token) is True
+    fake_request_body = 'token=xyzz0WbapA4vBCDEFasx0q6G&team_id=T1DC2JH3J&team_domain=testteamnow&channel_id=G8PSS9T3V&channel_name=foobar&user_id=U2CERLKJA&user_name=roadrunner&command=%2Fwebhook-collect&text=&response_url=https%3A%2F%2Fhooks.slack.com%2Fcommands%2FT1DC2JH3J%2F397700885554%2F96rGlfmibIGlgcZRskXaIFfN&trigger_id=398738663015.47445629121.803a0bc887a14d10d2c447fce8b6703c'
```
Comment (parberge) [inline tests/test_slack.py]: Might fit better in `test_data.py`

Comment (kamigerami) [reply tests/test_slack.py]: good point!

**Comment by kamigerami** [comment]
merging

**Pull request merged**